train_model.py

```diff
@@ -324,21 +332,46 @@ def get_params(self) -> dict:
     model_label = f"{datetime.now().strftime('%Y-%m-%d-%H:%M:%S')}_" + \
         f"{args.model}_{total_params}"
 
-    # Initialize model weights folders
-    save_folder = repo_root_dir / "weights" / model_label
-    save_folder.mkdir(parents=True, exist_ok=True)
-    print(f"\nSaving weights in {save_folder}")
+    # Initialize dataset folder
+    if args.dataset_dir is None:
+        data_folder = repo_root_dir / "data"
+    else:
+        data_folder = Path(args.dataset_dir)
+    data_folder.mkdir(parents=True, exist_ok=True)
+    print(f"\nUsing dataset cache directory {data_folder}")
+
+    # Initialize unique model save folder
+    if args.models_dir is None:
+        root_save_folder = repo_root_dir / "models" / model_label
+    else:
+        root_save_folder = Path(args.models_dir) / model_label
+    root_save_folder.mkdir(parents=True, exist_ok=False)
+    print(f"Saving model files in {root_save_folder}")
+
+    # Initialize weights folder. Shouldn't already exist or else run risk of
+    # previous model's weights being overwritten
+    if args.weights_dir is None:
+        weights_folder = root_save_folder / "weights"
+    else:
+        weights_folder = Path(args.weights_dir)
+    weights_folder.mkdir(parents=True, exist_ok=False)
+    print(f"Saving weight files in {weights_folder}")
+
+    # Create SummaryWriter to record logs for TensorBoard
+    if args.tboard_dir is None:
+        tboard_log_dir = repo_root_dir / "logs" / model_label
```

Comment:
can this be renamed tboard_log_folder to follow the pattern above? Namely,  data_folder, root_save_folder, and weights_folder?

Reply:
Great idea -- see [ec0abab](https://github.com/DRAGNLabs/301r_retnet/pull/21/commits/ec0abab52107cebfb5e75d2a7f8ff93a3b654727).